ui/components/app/confirm/info/row/address.tsx

```diff
@@ -1,41 +1,72 @@
-import React from 'react';
-import {
-  AvatarAccount,
-  Box,
-  Text,
-  AvatarAccountSize,
-} from '../../../../component-library';
+import { NameType } from '@metamask/name-controller';
+import React, { useState } from 'react';
+import { useSelector } from 'react-redux';
 import {
   AlignItems,
   BorderColor,
   Display,
   FlexDirection,
   TextColor,
 } from '../../../../../helpers/constants/design-system';
-import { shortenAddress } from '../../../../../helpers/utils/util';
-import Tooltip from '../../../../ui/tooltip/tooltip';
+import { getPetnamesEnabled } from '../../../../../selectors';
+import {
+  AvatarAccount,
+  AvatarAccountSize,
+  Box,
+  Text,
+} from '../../../../component-library';
+import NicknamePopovers from '../../../modals/nickname-popovers';
+import Name from '../../../name/name';
+import { useFallbackDisplayName } from './hook';
 
 export type ConfirmInfoRowAddressProps = {
   address: string;
 };
 
 export const ConfirmInfoRowAddress = ({
```

Comment:
FYI: This component is in use in production for Snaps.

While we are okay with using the petnames component, I am a bit worried that all of this is tied to the currently selected network. A Snap using this component may want to specify a chain ID to show petnames from a specific chain.

I'm wondering if this is something we could support? 🤔 

cc @matthewwalsh0

Reply:
Thanks for the comment @FrederikBolding. We have synced on Slack and agreed to add an optional variable to disable PetNames for snaps until they support explicitly passed `chainId`s.